JeMPI_Apps/JeMPI_Shared/src/main/java/org/jembi/jempi/shared/mapper/JsonToFhir.java

```diff
@@ -0,0 +1,120 @@
+import org.hl7.fhir.r4.model.*;
+import org.jembi.jempi.shared.models.*;
+import org.jembi.jempi.shared.utils.JsonFieldsConfig;
+import org.json.simple.JSONObject;
+import java.lang.reflect.Field;
+import java.util.List;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public final class JsonToFhir {
+
+    private static JsonFieldsConfig jsonFieldsConfig = new JsonFieldsConfig();
+
+    private JsonToFhir() throws Exception {
+    }
+
+    private static final Logger LOGGER = LogManager.getLogger(JsonToFhir.class);
+    private static String getFhirPath(final String fieldName) {
+        try {
+            for (int i = 0; i < jsonFieldsConfig.fields.size(); i++) {
+                JSONObject field = (JSONObject) jsonFieldsConfig.fields.get(i);
+                if (fieldName.equalsIgnoreCase((String) field.get("fieldName"))) {
+                    return (String) field.get("fhirPath");
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        return null;
+    }
+
+    private static void processField(final Patient patient, final String fieldValue, final String fhirPath) {
+        switch (fhirPath) {
+            case "Patient.identifier" -> {
+                Identifier identifier = new Identifier();
+                identifier.setValue(fieldValue);
+                patient.addIdentifier(identifier);
+            }
+            case "name.given" -> {
+                HumanName name = new HumanName();
+                name.addGiven(fieldValue);
+                patient.addName(name);
+            }
+            case "name.family" -> {
+                HumanName familyName = new HumanName();
+                familyName.setFamily(fieldValue);
+                patient.addName(familyName);
+            }
+            case "address.city" -> {
+                Address address = new Address();
+                address.setCity(fieldValue);
+                patient.addAddress(address);
+            }
+            case "birthDate" -> {
+                try {
+                    LocalDate date = LocalDate.parse(fieldValue, DateTimeFormatter.BASIC_ISO_DATE);
+                    String fhirDate = date.format(DateTimeFormatter.ISO_LOCAL_DATE);
+                    DateType birthDate = new DateType(fhirDate);
+                    patient.setBirthDate(birthDate.getValue());
+                } catch (DateTimeParseException e) {
+                    LOGGER.error("Failed to parse birth date: " + fieldValue, e);
+                }
+            }
+            default -> {
+                // to be implemented
+            }
+        }
+    }
+
+    public static String mapToPatientFhir(final String resourceId, final CustomDemographicData demographicData, final SourceId sourceId, final List<GoldenRecordWithScore> goldenRecordWithScoreList, final List<PatientRecordWithScore> patientRecordWithScoreList) {
```

Comment:
Refactoring : I would recommend having 3 functions : 
- private mapRecordToFhirFormat
- public mapPatientRecordToFhirFormat
- public mapGoldenRecordToFhirFormat

both mapPatientRecordToFhirFormat() and mapGoldenRecordToFhirFormat() would call mapRecordToFhirFormat() to format demographic data.